generic/src/decision.rs

```diff
@@ -555,12 +553,11 @@ pub fn decision(design: &mut Design) {
     let mut used_assigns = BTreeSet::new();
     for (decision, chain) in assign_chains.iter_disjoint(&decisions, &occurrences) {
         let (first_assign, last_assign) = (chain.first().unwrap(), chain.last().unwrap());
-        if cfg!(feature = "trace") {
-            eprintln!(">disjoint:");
+        tracing::trace_span!("disjoint").in_scope(|| {
             for &cell_ref in chain {
-                eprintln!("{}", design.display_cell(cell_ref));
+                tracing::trace!("{}", design.display_cell(cell_ref));
             }
-        }
+        });
```

Comment:
Can also use [tracing::enabled!](https://docs.rs/tracing/latest/tracing/macro.enabled.html) to skip the loop entirely. (The argument to `in_scope()` is executed unconditionally.)